### PR TITLE
Simplified union for FloatBox2D

### PR DESCRIPTION
**Pull request by sebastianbaginski**

### CODE REVIEW

File: source/math/FloatBox2D.ooc [hunk 1]
```diff
@@ -87,18 +87,10 @@ FloatBox2D: cover {
 		This new(left, top, width, height)
 	}
 	union: func ~box (other: This) -> This { // Rock bug: Union without suffix cannot be used because the C name conflicts with keyword "union"
-		left := Float minimum(this left, other left)
-		top := Float minimum(this top, other top)
-		width := Float maximum(0, (Float maximum(this right, other right) - left))
-		height := Float maximum(0, (Float maximum(this bottom, other bottom) - top))
-		This new(left, top, width, height)
+		This new(this leftTop minimum(other leftTop), this rightBottom maximum(other rightBottom))
```
Comment (marcusnaslund) [inline source/math/FloatBox2D.ooc]: Is a `Size` cover guaranteed to be nonnegative? The previous version made sure width and height were >= 0, that doesn't happen here.

Comment (marcusnaslund) [inline source/math/FloatBox2D.ooc]: Whether we should do that is a different question, though.

**Comment by marcusnaslund** [comment]
Letting this one wait until #765 is in.

File: source/math/FloatBox2D.ooc (hunk 1 above)
Comment (sebastianbaginski) [inline source/math/FloatBox2D.ooc]: I think this is the same change as in original issue : https://github.com/cogneco/ooc-kean/issues/753

Comment (marcusnaslund) [inline source/math/FloatBox2D.ooc]: There you changed only the one with a `Point` argument, here you changed both.

Comment (sebastianbaginski) [inline source/math/FloatBox2D.ooc]: That's true. I don't know really. Tests passed, everything seems to work, let's carry on

Comment (marcusnaslund) [inline source/math/FloatBox2D.ooc]: The discussion in #783 seems to show that size *should be* nonnegative. And you're right, this seems to work just fine now.

Comment (marcusnaslund) [inline source/math/FloatBox2D.ooc]: Also, if `width` or `height` are negative, then `leftTop` is not actually the left top, so.

**Pull request merged**